sdk/storage/Azure.Storage.Queues/src/QueueClient.cs

```diff
@@ -2227,6 +2239,38 @@ protected internal virtual QueueClient WithClientSideEncryptionOptionsCore(Clien
             }
         }
 
+        private async Task<QueueMessage[]> ToQueueMessagesWithInvalidMessageHandling(
+            IEnumerable<DequeuedMessageItem> dequeuedMessageItems,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            List<QueueMessage> queueMessages = new List<QueueMessage>();
+
+            foreach (var dequeuedMessageItem in dequeuedMessageItems)
+            {
+                try
+                {
+                    queueMessages.Add(QueueMessage.ToQueueMessage(dequeuedMessageItem, ClientConfiguration.MessageEncoding));
+                }
+                catch (FormatException)
+                {
+#pragma warning disable AZC0110 // DO NOT use await keyword in possibly synchronous scope.
+                    await ClientConfiguration.InvalidMessageHandler.RaiseAsync(
+                        new InvalidMessageEventArgs(
+                            queue: this,
+                            message: QueueMessage.ToQueueMessage(dequeuedMessageItem, QueueMessageEncoding.None),
+                            runSynchronously: !async,
```

Comment:
perhaps we need to change our helper to take async as the parameter to be internally consistent with other patterns we have.
cc @tg-msft

Reply:
yeah. analyzers were not happy about this as well.

Reply:
That feels redundant since we need the EventArgs anyway.  Should we just update the analyzers to learn this pattern (and suppress for now)?

Reply:
Filed https://github.com/Azure/azure-sdk-tools/issues/1357 to update the analyzers for this pattern.